### ISSUE TITLE
accessibility.signalOptions.delays not user friendly

### ISSUE THREAD

**Issue by isidorn** [microsoft/vscode#213618]
Testing #213344

Right now the structure of accessibility.signalOptions.delays settings is sort of leaking your implementation. It has a complex structure, and because of that is not shown in the settings editor, but only in the settings.json.

Can we simplify the structure of this setting so the settings UI editor supports showing it?

**Comment by meganrogge** [comment]
what would you suggest? I don't think adding 9 settings would be a good idea

**Comment by isidorn** [comment]
Any simplification as long as it shows up in the settings UI.
One example:

Get rid of `signalOptions`. Get rid of `volume`. Then have:

`accessibility.signals.debouncePositionChanges`
`accessibility.signals.delays`

**Comment by meganrogge** [comment]
That's what we had originally and @hediet suggested we introduce signal options.

**Comment by hediet** [comment]
The reason why I suggested `signalOptions` was that it stands out from all the other `signals`.
If you have 

* `accessibility.signals.progress`
* `accessibility.signals.delays`
* `accessibility.signals.save`
* ..., see ![Code_-_Insiders_TTU4YHIvUr](https://github.com/microsoft/vscode/assets/2931520/a4058711-302a-4b5d-a9d4-0791a855ad9e)

... it is very confusing which is a signal and which is a signal setting.

**Comment by isidorn** [comment]
Good point. So with your suggestion we are optimizing for `settings.json` users, and we are failing those that use the settings UI. Since the settings UI is the default experience my suggestion is to optimize for that.

Though I have no strong feeling either way.

**Comment by hediet** [comment]
I think the settings UI sorts alphabetically as well. So if we'd mix the settings with the signals, it would be confusing there as well.

Imagine if there are 3 interesting settings somewhere in this list of 29 audio cues:

![Code_-_Insiders_0wYFG0W0bL](https://github.com/microsoft/vscode/assets/2931520/a49263d7-f380-4f1b-931b-02b710fa2cf7)


---
We should just make sure `Signal Options` is editable in the settings editor:

![Code_-_Insiders_q2gSUET4Cr](https://github.com/microsoft/vscode/assets/2931520/bfb22b20-b230-46f7-a7f3-b5d394086421)

**Comment by meganrogge** [comment]
@rzhao271 is there a way we can do that? 

https://github.com/microsoft/vscode/blob/7ed51569a1bf53b40ea59268c482015770c1932e/src/vs/workbench/contrib/accessibility/browser/accessibilityConfiguration.ts#L174-L188

**Comment by rzhao271** [comment]
There's currently no support in the Settings editor for objects where the values could also be objects, even if the total object is just two levels deep.

**Comment by meganrogge** [comment]
I think we should close this as designed, do you agree @hediet ?

**Comment by isidorn** [comment]
Both options work for me.

**Comment by meganrogge** [comment]
I think we should probably destructure. Another argument for this is when I search for `debounce`, `debouncePositionChanges` doesn't even show up 🐛 

<img width="750" alt="Screenshot 2024-05-30 at 11 53 47 AM" src="https://github.com/microsoft/vscode/assets/29464607/5e920e05-7e90-4137-82b8-ed1d86c602d0">

**Comment by hediet** [comment]
Yes, we should destructure the object! But keep the namespace "signalOptions", e.g. "accessibility.signalOptions.delays", ....

**Comment by meganrogge** [comment]
verify that all of the `accessibility.signalOptions.*` settings work and are user friendly. Also verify that if you have settings of the old structure and reload the window, they get migrated correctly.

**Comment by isidorn** [comment]
This works better now, thank you! Adding verified label.
